### DISCUSSION TITLE
Should escape in menus clear search?

### DISCUSSION THREAD

**Discussion by tomasklaen** [tomasklaen/uosc#718] [General]
When I use menus and I search something there (like keybinds), I expect escape to close it, but that just clears the search, and leaves me confused for a second.

I think escape should always close the menu, and maybe shift+escape clear or close. Or rather `ctrl` or `shift` + `backspace` to clear the search and no double function buttons?

**Comment by FichteFoll** [comment]
> I think escape should always close the menu

+1

ctrl+backspace is commonly used to delete the previous word, not the full line preceding the caret, so I'm against using this as it would be inconsistent. shift+backspace usually does the same as just backspace.

When I want to clear a search field, I usually use shfit+home or ctrl+a and then backspace. That's just one key chord more but works universally.

Another option would be to make reopening the menu simply clear the search (when the menu can be opened with a key binding).

**Comment by tomasklaen** [comment]
I'm currently leaning to:

- `escape` - close only
- `ctrl+backspace` - delete to first non-word char
- `shift+backspace` - clear query

**Reply by christoph-heinrich** [reply]
`alt` could also be used as a modifier, but your suggestion already seems reasonable.

**Comment by tomasklaen** [comment]
#719
